### PR TITLE
Feature add start stop

### PR DESCRIPTION
**Pull request by mlohvynenko**

### CODE REVIEW

File: src/communication/channelmanager.cpp [hunk 1]
```diff
@@ -86,23 +88,24 @@ Error ChannelManager::Connect()
...
+    }
 
-        if (auto err = mTransport->Close(); !err.IsNone()) {
-            return err;
-        }
+    if (auto err = mTransport->Close(); !err.IsNone()) {
+        return err;
     }
 
+    lock.Unlock();
```
Comment (MykolaSolyanko) [inline src/communication/channelmanager.cpp:105]: maybe use unamed scope `{}` and use LockGuard?

File: src/communication/channelmanager.cpp [hunk 2]
```diff
@@ -54,6 +54,8 @@ RetWithError<ChannelItf*> ChannelManager::CreateChannel(uint32_t port)
         return {nullptr, err};
     }
 
+    lock.Unlock();
```
Comment (MykolaSolyanko) [inline src/communication/channelmanager.cpp:57]: maybe also better to use LockGuard and unnamed block?

File: src/clocksync/clocksync.cpp [hunk 3]
```diff
@@ -128,14 +149,6 @@ void ClockSync::Unsubscribe(ClockSyncSubscriberItf& subscriber)
 
 ClockSync::~ClockSync()
```
Comment (MykolaSolyanko) [inline src/clocksync/clocksync.cpp:150]: maybe remove empty destructor

File: west.yml [hunk 4]
```diff
@@ -21,15 +21,18 @@ manifest:
     - name: zephyrproject-rtos
       url-base: https://github.com/zephyrproject-rtos
 
+    - name: mlohvynenko
```
[the remark this answers is not in the document]
Comment (mlohvynenko) [reply west.yml:24]: yeap, once core lib cpp PR is merged into the  feature branch

File: src/communication/channelmanager.cpp (hunk 2 above)
Comment (mlohvynenko) [reply src/communication/channelmanager.cpp:57]: In this case I'd like to  keep as it.
The main reason is the channel variable that's declared only if mChannels doesn't contain the desired port.

It's not a big deal to declare it as nullptr before proposed LockGuard scope.
But I don't see much benefits in this change.

**Review by MykolaSolyanko — APPROVED** [review]

File: tests/iamclient/src/main.cpp [hunk 5]
```diff
@@ -65,9 +65,18 @@ static void InitIAMClient(iamclient_fixture* fixture, const aos::NodeInfo& nodeI
...
 
+    err = fixture->mIAMClient->Start();
+    zassert_true(err.IsNone(), "Can't start IAM client: %s", utils::ErrorToCStr(err));
+
     fixture->mIAMClient->OnClockSynced();
 }
 
+static void StopIAMClient(iamclient_fixture* fixture)
+{
+    auto err = fixture->mIAMClient->Stop();
```
Comment (al1img) [inline tests/iamclient/src/main.cpp:76]: No need for new function. Put stop before test I am client is recreated

File: west.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (al1img) [reply west.yml:24]: merged. We should switch all components to develop branch.

File: src/communication/channel.cpp [hunk 6]
```diff
@@ -33,11 +33,14 @@ Error Channel::Connect()
 
 Error Channel::Close()
 {
-    LockGuard lock {mMutex};
+    {
```
Comment (al1img) [inline src/communication/channel.cpp:36]: Unclear, why?

File: src/communication/channel.cpp [hunk 7]
```diff
@@ -33,11 +33,14 @@ Error Channel::Connect()
 
 Error Channel::Close()
 {
-    LockGuard lock {mMutex};
```
Comment (al1img) [inline src/communication/channel.cpp:36]: Ulock typo in commit message.

File: src/clocksync/clocksync.cpp [hunk 8]
```diff
@@ -71,19 +71,37 @@ Error ClockSync::Init(ClockSyncSenderItf& sender)
 
 Error ClockSync::Start()
 {
-    LockGuard lock {mMutex};
+    {
```
Comment (al1img) [inline src/clocksync/clocksync.cpp:74]: Why?

File: src/clocksync/clocksync.cpp [hunk 9]
```diff
@@ -71,19 +71,37 @@ Error ClockSync::Init(ClockSyncSenderItf& sender)
...
+{
+    {
+        LockGuard lock {mMutex};
+
+        LOG_DBG() << "Stop";
+
+        mClose = true;
+    }
+
+    mCondVar.NotifyOne();
```
Comment (al1img) [inline src/clocksync/clocksync.cpp:97]: Could be under lock

File: src/main.cpp [hunk 10]
```diff
@@ -30,6 +30,19 @@ using namespace aos::zephyr;
 
 int main(void)
 {
+    atexit([]() {
```
Comment (al1img) [inline src/main.cpp:33]: Is it compiled for target platforms?
If so, move it before apps.Init

Comment (mlohvynenko) [reply src/main.cpp:33]: Successfully compiled for rpi5 board

File: src/smclient/smclient.cpp [hunk 11]
```diff
@@ -30,17 +30,17 @@ static google_protobuf_Timestamp TimestampToPB(const Time& time)
...
+    pbMonitoringData.download         = monitoringData.mDownload;
+    pbMonitoringData.upload           = monitoringData.mUpload;
+    pbMonitoringData.partitions_count = monitoringData.mDisk.Size();
+    pbMonitoringData.has_timestamp    = true;
+    pbMonitoringData.timestamp        = TimestampToPB(timestamp);
 
     for (size_t i = 0; i < monitoringData.mDisk.Size(); i++) {
-        utils::StringFromCStr(pbMonitoringData.disk[i].name) = monitoringData.mDisk[i].mName;
-        pbMonitoringData.disk[i].used_size                   = monitoringData.mDisk[i].mUsedSize;
+        utils::StringFromCStr(pbMonitoringData.partitions[i].name) = monitoringData.mDisk[i].mName;
```
Comment (al1img) [inline src/smclient/smclient.cpp:42]: We need to rename mDisk -> mPartitions in monitoring data as well.

Comment (mlohvynenko) [reply src/smclient/smclient.cpp:42]: Will be done in a separate PR

**Pull request merged**